app/models/mixins/authentication_mixin.rb

```diff
@@ -2,10 +2,49 @@ module AuthenticationMixin
   extend ActiveSupport::Concern
 
   included do
+    # There are some dirty, dirty Ruby/Rails reasons why this method needs to
+    # exist like this, and I will try and explain:
+    #
+    # First, we need the extra +.where+ here because if this is used in a
+    # nested SELECT statement (aka: virtual_delegate), the +resource_type+
+    # check is dropped on the floor and not included in the subquery.  As a
+    # result, it will just pickup the first record to match the
+    # relationship_id column and that could be associated with any OTHER object
+    # with an Authentication record.
+    #
+    # For example, you get back an authentication record for an EMS record when
+    # you are looking up on for a Host.
+    #
+    # Secondly, the reason this method exists, and is FIRST (prior to the
+    # +has_one+ that uses it) is it needs to be defined prior to the
+    # relationship method that calls it.  Also, it needs to be a method so the
+    # proper local variable, +authentication_mixin_relation+, can be defined
+    # with the class that the +resource_type+ needs to match and remain in
+    # scope for the Proc below.  If it is a class variable or done in other
+    # fashion, it will be overwritten whenever this module is included
+    # elsewhere.
+    #
+    # Finally, the use of a prepared statement is also for some reason required
+    # over the Hash syntax since otherwise the following is ERROR is produced
+    # when doing a nested SELECT:
```

Comment:
What was the Hash syntax that you tried?  Presuming you tried `.where(:authentications => {:resource_type => name})`

Reply:
It was just:

```ruby
.where({:resource_type => authentication_mixin_relation)
```

Though the error suggests it wasn't playing nicely with the `binds` and dropping the ones in the subquery, not so much the syntax.

Reply:
FYI:  I further confirmed the above doesn't work, and gave a bit of a report in this comment:

https://github.com/ManageIQ/manageiq/pull/20494/files#r480396758

Going to resolve this one.